discovery/discovery_test.go

```diff
@@ -248,13 +248,15 @@ func TestDiscoverEndpoints(t *testing.T) {
 
 	for i, tt := range tests {
 		httpDo = tt.do
+		httpDoInsecureTls = tt.do
```

Comment:
This test needs to be extended a bit, we are using insecure flags unconditionally, so discovery will never use `httpDo`.